python_modules/dagster/dagster/core/execution_plan/simple_engine.py

```diff
@@ -301,7 +304,11 @@ def _create_step_output_event(step_context, step_output_value, intermediates_man
             value=value,
         )
 
-        return DagsterEvent.step_output_event(
+        yield DagsterEvent.step_materialization(
```

Comment:
i think we should only do this when we are persisting intermediates somewhere and it should probably be its own event type.